File: backend/app/schemas/history.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class HistoryEntry(BaseModel):
    id: str
    topic: str
    topic_type: str  # "letter" | "motion_sign"
    status: str  # pass, fail, no_attempt_detected
    correct: Optional[bool] = None
    confidence: Optional[float] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class PracticeHistoryResponse(BaseModel):
    learner_id: str
    total_sessions: int
    scored_sessions: int
    accuracy_percent: Optional[float] = None
    entries: list[HistoryEntry]

File: Team_3_Sign_Language_AI-Karishma-milestone3/frontend/src/types/history.ts
// Mirrors backend/app/schemas/history.py — the raw, session-by-session
// practice log behind the Practice History page, distinct from
// LearnerAnalytics' aggregated per-letter/per-day view.

import { TopicType } from './analytics';

export type PracticeStatus = 'pass' | 'fail' | 'no_attempt_detected';

export interface HistoryEntry {
  id: string;
  topic: string;
  topic_type: TopicType;
  status: PracticeStatus;
  correct: boolean | null;
  confidence: number | null;
  created_at: string | null;
}

export interface PracticeHistory {
  learner_id: string;
  total_sessions: number;
  scored_sessions: number;
  accuracy_percent: number | null;
  entries: HistoryEntry[];
}
